scripts/maint/annotate_ifdef_directives

```diff
@@ -26,14 +26,33 @@ import re
 # Any block with fewer than this many lines does not need annotations.
 LINE_OBVIOUSNESS_LIMIT = 4
 
+# Maximum line width.
+LINE_WIDTH=80
+
 class Problem(Exception):
     pass
 
-def commented_line(fmt, argument):
+def commented_line(fmt, argument, maxwidth=LINE_WIDTH):
     """
-    Return fmt%argument, for use as a commented line.
+    Return fmt%argument, for use as a commented line.  If the line would
+    be longer than maxwidth, truncate argument.
+
+    Requires that fmt%"..." will fit into maxwidth characters.
     """
-    return fmt % argument
+    result = fmt % argument
+    if len(result) <= maxwidth:
```

Comment:
I think this will allow a 80-character line. doc/HACKING/Codingstyle.md says 79 characters per line, and checkSpaces.pl seems to enforce 79.

Reply:
Should we have a test case for 79 and 80 character lines?
Or is this script too small to have test cases?

Reply:
> Should we have a test case for 79 and 80 character lines?
> Or is this script too small to have test cases?

I think that's a great idea to have a test case for 79 vs 80 character lines.